src/environmentd/src/http.rs

```diff
@@ -217,8 +219,19 @@ impl Server for HttpServer {
     fn handle_connection(&self, conn: Connection) -> ConnectionHandler {
         let router = self.router.clone();
         let tls_config = self.tls.clone();
-        let conn = TokioIo::new(conn);
+        let mut conn = TokioIo::new(conn);
         Box::pin(async {
+            let direct_peer_addr = conn
+                .inner()
+                .peer_addr()
+                .expect("http connection must have a peer ip");
```

Comment:
This `expect` worries me. I don't know in exactly what cases `peer_addr` can return an error, but I also don't think I want to find out via a panic in production! I think you should just be able to use `?` to return the error here. Ideally with a bit of context:

```suggestion
            let direct_peer_addr = conn
                .inner()
                .peer_addr()
                .context("fetching peer addr")?;
```

That should forcibly terminate the connection from the client side, but log an error on the server side.